torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -3174,11 +3174,38 @@ struct to_ir {
           return dim + 1;
         }
       }
-      TypePtr type_hint = OptionalType::ofTensor();
-      if (subscript_expr.kind() == TK_NONE) {
-        type_hint = NoneType::get();
+      Value* index;
+      if (subscript_expr.kind() == TK_LIST_LITERAL) {
+        // Accept list literal as subscript but convert it to a Tensor
+        // since it's equivalent to indexing with Tensor.
+        // Advanced indexing using list:
+        // @torch.jit.script
+        // def f(x):
+        //     return x[[0, 1, 5]]  # or
+        //     return x[[0, 1], [0, 1]]  # or
+        //     return x[[[0, 1], [0, 1]], [[0, 1], [0, 1]]]
+        // Statements above are equivalent to advanced indexing using Tensor:
+        // @torch.jit.script
+        // def f(x):
+        //     return x[torch.tensor([0, 1, 5])]  # or
+        //     return x[torch.tensor([0, 1]), torch.tensor([0, 1])]  # or
+        //     return x[torch.tensor([[0, 1], [0, 1]]), torch.tensor([[0, 1],
+        //     [0, 1]])]
+        auto ll = emitExpr(subscript_expr);
+        Value* dtype = graph->insertNode(graph->createNone())->output();
```

Comment:
Do we need to explicitly create these arguments? From the schema it looks like they are defaulted.

Also, you should use schema-driven insertion (`Graph::insert`) instead of a raw `create` + `insertNode`. `Graph::insert` which will do some bookkeeping to make sure you are inserting the node correctly, and might give you the default args for free.

Reply:
Nice this is indeed what I was looking for :D !